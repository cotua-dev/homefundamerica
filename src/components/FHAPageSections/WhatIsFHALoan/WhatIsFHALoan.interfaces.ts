export interface FetchWhatIsFHALoanImageQuery {
    whatIsFHALoanImage: {
        publicURL: string;
    };
}
