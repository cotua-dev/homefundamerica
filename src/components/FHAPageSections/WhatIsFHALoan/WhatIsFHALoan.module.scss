@import '../../../styles/variables';

section#whatIsFHALoanSection {
    height: 60vh;

    @media screen and (max-width: 950px) {
        height: 100%;
        flex-direction: column-reverse;
    }
}
