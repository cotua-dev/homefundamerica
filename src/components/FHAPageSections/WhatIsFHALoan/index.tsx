import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import {
    useStaticQuery,
    graphql,
} from 'gatsby';
import { whatIsFHALoanSection } from './WhatIsFHALoan.module.scss';
import * as sharedStyles from '../../sharedStyles/shared.module.scss';
import { FetchWhatIsFHALoanImageQuery } from './WhatIsFHALoan.interfaces';

export const WhatIsFHALoan: FunctionComponent = (): ReactElement => {
    const queryResult = useStaticQuery<FetchWhatIsFHALoanImageQuery>(graphql`
        query {
            whatIsFHALoanImage: file(name: {eq: "AdobeStock_130155999"}, extension: {eq: "jpg"}) {
                publicURL
            }
        }
    `);

    return (
        <section
            id={whatIsFHALoanSection}
            className={sharedStyles.loanPositive}
        >
            <div className={sharedStyles.contentWrapper}>
                <h2>{`What is an FHA Loan?`}</h2>
                <p>
                    {`An FHA Loan is a government mortgage insured by the Federal
                    Housing Administration (FHA). FHA Loans offer borrowers relaxed
                    credit requirements and more flexible underwriting guidelines
                    than conventional loans. In additions to purchase loans, refinance
                    programs are available for FHA borrowers, non-FHA borrowers, and
                    homeowners interested cash-out options. Up to 85% of the home's
                    current value can be taken out in cash with a refinance.`}
                </p>
            </div>
            <div className={sharedStyles.imageWrapper}>
                <img
                    src={queryResult?.whatIsFHALoanImage.publicURL}
                    alt="What is an FHA loan image"
                />
            </div>
        </section>
    );
}
