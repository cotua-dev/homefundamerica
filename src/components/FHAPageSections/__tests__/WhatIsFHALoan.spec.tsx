import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { WhatIsFHALoan } from '../WhatIsFHALoan';

describe('WhatIsFHALoan Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<WhatIsFHALoan/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
