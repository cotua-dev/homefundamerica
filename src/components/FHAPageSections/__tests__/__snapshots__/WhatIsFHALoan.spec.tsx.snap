// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhatIsFHALoan Component renders 1`] = `
<section
  id="whatIsFHALoanSection"
>
  <div>
    <h2>
      What is an FHA Loan?
    </h2>
    <p>
      An FHA Loan is a government mortgage insured by the Federal
                    Housing Administration (FHA). FHA Loans offer borrowers relaxed
                    credit requirements and more flexible underwriting guidelines
                    than conventional loans. In additions to purchase loans, refinance
                    programs are available for FHA borrowers, non-FHA borrowers, and
                    homeowners interested cash-out options. Up to 85% of the home's
                    current value can be taken out in cash with a refinance.
    </p>
  </div>
  <div>
    <img
      alt="What is an FHA loan image"
    />
  </div>
</section>
`;
